pkg/controller/atlasdatabaseuser/atlasdatabaseuser_controller.go

```diff
@@ -18,11 +18,11 @@ package atlasdatabaseuser
 
 import (
 	"context"
-	"errors"
 	"fmt"
 
 	"go.uber.org/zap"
 	corev1 "k8s.io/api/core/v1"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
```

Comment:
very small nit: the convention in upstream is `apierrors`, see the kube code base. (really not that urgent)

Reply:
Uhm I think I better separate that refactor on its own. It is not related to the main refactor and this name is used on several other files that will distract from the main refactor focus here.

Reply:
https://github.com/mongodb/mongodb-atlas-kubernetes/pull/1804